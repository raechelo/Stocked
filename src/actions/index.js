export const isLoading = bool => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const setError = message => ({
  type: 'SET_ERROR',
  message
});

export const setRecipes = recipes => ({
  type: 'SET_RECIPES',
  recipes
});

export const toggleFavorite = id => ({
  type: 'TOGGLE_FAVORITE',
  id
});

export const filterRecipes = input => ({
  type: 'FILTER_RECIPES',
  input
});

export const setIngredients = ing => ({
  type: 'SET_INGREDIENTS',
  ing
});

export const setResults = results => ({
  type: 'SET_RESULTS',
  results
})