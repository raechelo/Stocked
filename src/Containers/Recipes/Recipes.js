import React, { Component } from 'react'
import RecipeCard from '../../Components/RecipeCard/RecipeCard';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { addFavorite, deleteFavorite } from '../../actions/index';
import Loading from '../../Components/Loading/Loading';

export class Recipes extends Component {

  favoriteRecipe = (fave) => {
    if (this.props.favorites.includes(fave)) {
      this.props.deleteFavorite(fave.idMeal)
    } else {
      this.props.addFavorite(fave)
    }
  }
  
  render () {
    let { recipes } = this.props
    let displayCards = recipes.map(r => (<RecipeCard {...r} key={r.idMeal} addFavorite={this.favoriteRecipe} />))
    return (
      <section className="Recipes">
        { this.props.isLoading ? <Loading /> : displayCards }
      </section>
    )
  }
}

Recipes.propTypes = {
  recipes: PropTypes.array,
  isLoading: PropTypes.bool
}

export const mapStateToProps = state => ({
    recipes: state.recipes,
    isLoading: state.isLoading,
    favorites: state.favorites
});

export const mapDispatchToProps = dispatch => ({
  addFavorite: favorite => dispatch(addFavorite(favorite)),
  deleteFavorite: id => dispatch(deleteFavorite(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(Recipes);
  