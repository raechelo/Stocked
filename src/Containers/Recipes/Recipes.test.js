import React from 'react';
import { Recipes, mapDispatchToProps, mapStateToProps } from './Recipes';
import { shallow } from 'enzyme';
import { addFavorite } from '../../actions/index';

describe('Recipes', () => {

  let wrapper;
  let mockRecipes = [{title: 'Mango Salsa', id: 1}, {title: 'Bacon Pancakes', id: 2}, {title: 'Jalapeno Grilled Cheese', id: 3}]

  beforeEach(() => {
    wrapper = shallow(<Recipes recipes={mockRecipes} />);
  });

  it('should match the snapshot with all the data paseed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  
  it('should return an object with the recipes array', () => {
    const mockState = {
      recipes: [{name:'Avocados', id: 1}]
    }
    const expected = {
      recipes: [{name:'Avocados', id: 1}]
    }

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

  it('should return an object with the isLoading value', () => {
    const mockState = {
      isLoading: true
    }
    const expected = {
      isLoading: true
    }

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  let mockDispatch = jest.fn();

  it('should dispatch addFavorites', () => {
    const mockState = {
      favorites: ['Avocados', 'Mangoes']
    }
    const actionToDispatch = addFavorite(mockState);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.addFavorite(mockState);
    
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});