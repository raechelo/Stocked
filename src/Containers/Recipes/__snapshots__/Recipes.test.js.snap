// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes should match the snapshot with all the data paseed in 1`] = `ShallowWrapper {}`;
